src/snmalloc/ds/flaglock.h

```diff
@@ -56,24 +57,20 @@ namespace snmalloc
     }
 
   private:
-    using ThreadIdentity = int const*;
-
+    using ThreadIdentity = size_t;
```

Comment:
Should this be a type provided by the PAL?  I can imagine some platforms might want to use a pointer to a thread control block, for example, rather than an integer.